File: JavaScript/JavaScript/SelectionSort.js
/**
 * 選択ソート
 * @param list リスト
 */
function selectionSort(list) {
  for (let i = 0; i < list.length; i++) {
      let minj = i;
      for (j = i; j < list.length; j++) {
          if (list[j] < list[minj]) {
              minj = j;
          }
      }
      let beSortedValue = list[i];
      list[i] = list[minj];
      list[minj] = beSortedValue;
  }
  return list;
}


console.log(selectionSort([5, 2, 4, 6, 1, 3]));

File: Dart/SelectionSort.dart
void main() {
  
  /**
   * 選択ソート
   * @param list リスト
   */
  List<int> selectionSort(List<int> list) {
    final n = list.length;
    for (var i = 0; i < n; i++) {
      var minj = i;
      for (var j = minj; j < n; j++) {
        if (list[minj] > list[j]) {
          minj = j;
        }
      }
      var beSortedValue = list[i];
      list[i] = list[minj];
      list[minj] = beSortedValue;
    }
    return list;
  }
  print(selectionSort([4, 2, 5, 6, 1, 3]));
}
